omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/HelpersMethods.java

```diff
@@ -0,0 +1,140 @@
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isClickable;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+public class HelpersMethods extends EspressoMethods {
+    public static boolean isVisible(Matcher<View> matcher) {
+        try {
+            onView(matcher).check(matches(isDisplayed()));
+            return true;
+        } catch (Exception e) {
+            // View is not in hierarchy
+            return false;
+        }
+    }
+
+    public static boolean isUIObjectClickable(Matcher<View> matcher) {
+        try {
+            onView(matcher).check(matches(isClickable()));
+            return true;
+        } catch (NoMatchingViewException e) {
+            // View is not in hierarchy
+            return false;
+        }
+    }
+
+    public static void clickAction(Matcher<View> matcher) {
+        onView(matcher).perform(click());
+    }
+
+    public static boolean isFirstObjectToTheLeftOfSecondObject(Matcher<View> matcher1, Matcher<View> matcher2) {
+        try {
+            onView(matcher1).check(isLeftOf(matcher2));
+            return true;
+        } catch (NoMatchingViewException e) {
+            // View is not in hierarchy
+            return false;
+        }
+    }
+
+    public static boolean isFirstObjectToTheRightOfSecondObject(Matcher<View> matcher1, Matcher<View> matcher2) {
+        try {
+            onView(matcher1).check(isRightOf(matcher2));
+            return true;
+        } catch (NoMatchingViewException e) {
+            // View is not in hierarchy
+            return false;
+        }
+    }
+
+    public static boolean isFirstObjectAboveOfSecondObject(Matcher<View> matcher1, Matcher<View> matcher2) {
+        try {
+            onView(matcher1).check(isAbove(matcher2));
+            return true;
+        } catch (NoMatchingViewException e) {
+            // View is not in hierarchy
+            return false;
+        }
+    }
+
+    public static boolean isFirstObjectBelowOfSecondObject(Matcher<View> matcher1, Matcher<View> matcher2) {
+        try {
+            onView(matcher1).check(isBelow(matcher2));
+            return true;
+        } catch (NoMatchingViewException e) {
+            // View is not in hierarchy
+            return false;
+        }
+    }
+
+    private static String getRandomString() {
+        final String AB = Constants.ALPHANUMERIC_RANDOM_STRING;
+
+        SecureRandom random = new SecureRandom();
+        StringBuilder stringBuilder = new StringBuilder(5);
+
+        for (int i = 0; i < 5; i++) {
+            stringBuilder.append(AB.charAt(random.nextInt(AB.length())));
+        }
+
+        return stringBuilder.toString();
+    }
+
+    public static void typeNewItemAction(int xItemsAdded, int xPositionToAddNewText) {
+        for (int i = 0; i <= xItemsAdded; i++) {
```

Comment:
You could use "i < xItemsAdded", because when you call this method you have to use an argument which the desired numbers to add -1. And this could be misleading.

Reply:
Done.